tgui/packages/tgui/interfaces/LibraryConsole.js

```diff
@@ -0,0 +1,988 @@
+import { map, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { classes } from 'common/react';
+import { marked } from 'marked';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Dropdown, Input, Modal, NoticeBox, NumberInput, LabeledList, Section, Stack, Table } from '../components';
+import { Window } from '../layouts';
+import { sanitizeText } from "../sanitize";
+
+export const LibraryConsole = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    show_dropdown,
+    display_lore,
+  } = data;
+  return (
+    <Window
+      theme={display_lore ? "spookyconsole" : ""}
+      title="Library Terminal"
+      width={880}
+      height={520}>
+      <PopoutMenu />
+      <Box fillPositionedParent left={show_dropdown ? "164px" : "50px"}>
+        <Window.Content>
+          <PageDisplay pb="10px" />
+        </Window.Content>
+      </Box>
+    </Window>
+  );
+};
+
+export const PopoutMenu = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    screen_state,
+    show_dropdown,
+    display_lore,
+  } = data;
+  return (
+    <Section fill maxWidth={show_dropdown ? "150px" : "36px"}>
+      <Stack vertical fill>
+        <Stack.Item>
+          <Button
+            fluid
+            fontSize="13px"
+            onClick={() => act('toggle_dropdown')}
+            icon={show_dropdown === 1 ? 'chevron-left' : 'chevron-right'}
+            tooltip={!show_dropdown && 'Expand'}
+            content={!!show_dropdown && 'Collapse'} />
+        </Stack.Item>
+        <PopoutEntry
+          id={1}
+          icon="list"
+          text="Inventory" />
+        <PopoutEntry
+          id={2}
+          icon="calendar"
+          text="Checkout" />
+        <PopoutEntry
+          id={3}
+          icon="server"
+          text="Archive" />
+        <PopoutEntry
+          id={4}
+          icon="upload"
+          text="Upload" />
+        <PopoutEntry
+          id={5}
+          icon="print"
+          text="Print" />
+        {!!display_lore && (
+          <PopoutEntry
+            id={6}
+            icon="question"
+            text={screen_state === 6 ? "Gur Fbeprere": "Forbidden Lore"}
+            color="black"
+            font="copperplate" />
+        )}
+      </Stack>
+    </Section>
+  );
+};
+
+export const PageDisplay = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    screen_state,
+  } = data;
+
+  if (screen_state === 1) {
+    return (
+      <Inventory />
+    );
+  }
+  if (screen_state === 2) {
+    return (
+      <Checkout />
+    );
+  }
+  if (screen_state === 3) {
+    return (
+      <Archive />
+    );
+  }
+  if (screen_state === 4) {
+    return (
+      <Upload />
+    );
+  }
+  if (screen_state === 5) {
+    return (
+      <Print />
+    );
+  }
+  if (screen_state === 6) {
+    return (
+      <Forbidden />
+    );
+  }
```

Comment:
For "cool syntax" points, add eslint-ignore if linter complains.

```suggestion
  return (
    screen_state === 1 ? <Inventory /> :
    screen_state === 2 ? <Checkout /> :
    screen_state === 3 ? <Archive /> :
    screen_state === 4 ? <Upload /> :
    screen_state === 5 ? <Print /> :
    screen_state === 6 ? <Forbidden /> :
    null
  );
```